docker-app/Dockerfile

```
@@ -3,7 +3,7 @@
 ##########################
 
 # pull a builder image, the same as the base
-FROM python:3.10.8-slim-buster AS build
+FROM python:3.10-slim AS build
```

```suggestion
FROM python:3.10-8-slim-buster AS build
```

Let's keep in this PR without such change and have a follow-up where we pin to `3.10-slim-bookworm`. I would like to stick to specific Debian version for the same reasons I have described in one of the other PRs.

Btw, we managed to update almost all dependencies and soon we will be able to move to python 3.12+.

---

Look at the CI:
```
 > [qgis  7/15] RUN if [ "1" = "1" ]; then apt install -y gdbserver valgrind; fi:
0.051 
0.054 Reading package lists...
0.067 Building dependency tree...
0.068 Reading state information...
0.070 Package valgrind is not available, but is referred to by another package.
0.070 This may mean that the package is missing, has been obsoleted, or
0.070 is only available from another source
0.070 
0.070 E: Unable to locate package gdbserver
0.070 E: Package 'valgrind' has no installation candidate
```

Exactly why I don't want to use `latest` tags :)

---

OK, i will just remove this from the PR

---

The valgrind was a mistake of mine and should be fixed now.

---

If everything passes, the PR should be ready now.